src/pool/mempool.rs

```diff
@@ -58,35 +58,34 @@ impl AccountManager {
         Self { accounts }
     }
 
-    pub fn start<SP>(&self, rt_handle: &Handle, eth_provider: &'static EthDataProvider<SP>)
+    pub fn start<SP>(&self, rt_handle: &Handle, eth_client: &'static EthClient<SP>)
     where
-        SP: starknet::providers::Provider + Send + Sync + 'static,
+        SP: starknet::providers::Provider + Send + Sync + Clone + 'static,
     {
         let accounts = self.accounts.clone();
 
         rt_handle.spawn(async move {
             loop {
                 for address in &accounts {
-                    Self::process_transaction(address, eth_provider);
+                    Self::process_transaction(address, eth_client);
                 }
 
                 tokio::time::sleep(Duration::from_secs(1)).await;
             }
         });
     }
 
-    fn process_transaction<SP>(address: &Felt, eth_provider: &EthDataProvider<SP>)
+    fn process_transaction<SP>(address: &Felt, eth_client: &EthClient<SP>)
     where
-        SP: starknet::providers::Provider + Send + Sync + 'static,
+        SP: starknet::providers::Provider + Send + Sync + Clone + 'static,
```

Comment:
same

Reply:
Yes it because when we do `eth_client.mempool()` we go to the function here

https://github.com/kkrt-labs/kakarot-rpc/blob/95e0a0f7131161dd7db4c7b51889e31164923b44/src/client/mod.rs#L62-L65

and in this implementation we have the constraint 

https://github.com/kkrt-labs/kakarot-rpc/blob/95e0a0f7131161dd7db4c7b51889e31164923b44/src/client/mod.rs#L26-L29

So here we need to have the `Clone` constraint...